tests/unit/test_graph_runnable_task.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from dataclasses import dataclass
+from dbt.task.runnable import GraphRunnableTask
+from typing import AbstractSet, Any, Dict, Optional
+
+
+@dataclass
+class MockArgs:
+    """Simple mock args for us in a runnable task"""
+
+    state: Optional[Dict[str, Any]] = None
+    defer_state: Optional[Dict[str, Any]] = None
+    write_json: bool = False
+
+
+@dataclass
+class MockConfig:
+    """Simple mock config for use in a RunnableTask"""
+
+    threads: int = 1
+    target_name: str = "mock_config_target_name"
+
+
+class CustomRunnableTask(GraphRunnableTask):
+    did_cancel: bool = False
+
+    def run_queue(self, pool):
+        """Override `run_queue` to raise a system exit"""
+        raise SystemExit()
+
+    def _cancel_connections(self, pool):
+        """Override `_cancel_connections` to track whether it was called"""
+        self.did_cancel = True
+
+    def get_node_selector(self):
+        """This is an `abstract_method` on `GraphRunnableTask`, thus we must implement it"""
+        return None
+
+    def defer_to_manifest(self, adapter, selected_uids: AbstractSet[str]):
+        """This is an `abstract_method` on `GraphRunnableTask`, thus we must implement it"""
+        return None
+
+
+def test_graph_runnable_task_cancels_connection_on_system_exit():
+    task = CustomRunnableTask(args=MockArgs(), config=MockConfig(), manifest=None)
+
+    with pytest.raises(SystemExit):
```

Comment:
while we're here let's also test for [KeyboardInterrupt](https://github.com/dbt-labs/dbt-core/pull/8994/files#diff-f01af78b7eaa1bacc301afa75eea4c7cdbe03064b3913bc2ce3f6bf01d0bc125R374)?